pphysx3d/src/scene/mod.rs

```diff
@@ -29,24 +31,103 @@ impl PhysicsScene {
     }
 
     /// Updates the physics in the scene, such as collisions
-    pub fn update(&mut self, time_step: f32) {
+    pub fn update(&mut self) {
         // Physics loop
 
         // Detect collisions
         let collision_pairs = broad_phase(&self.objects);
         let manifolds = narrow_phase(&self.objects, &collision_pairs);
 
-        // Resolve Collisions & Apply impulse
+        // Resolve collisions & apply impulse + friction
+        for (i, manifold) in manifolds.iter().enumerate() {
+            if manifold.colliding {
+                //&self.objects[i].add_force(Vector3::new(0., 0., 0.)); // this would be how to add new forces but we don't do that right here
+                // I DON'T KNOW IF THIS IS HOW RUST IS SUPPOSED TO BE WRITTEN BTW BUT CARGO IS HAPPY
+                let index = &collision_pairs[i];
+                // Calculate variables for readability: (unnecessary?)
+                let (
+                    mass_1,
+                    mass_2,
+                    restitution_1,
+                    restitution_2,
+                    e,
+                    j,
+                    t,
+                    mut jt,
+                    friction,
+                    velocity_1,
+                    velocity_2,
+                    v_r,
+                ): (
+                    f32,
+                    f32,
+                    f32,
+                    f32,
+                    f32,
+                    f32,
+                    Vector3<f32>,
+                    f32,
+                    f32,
+                    Vector3<f32>,
+                    Vector3<f32>,
+                    Vector3<f32>,
+                );
+                {
```

Comment:
partitioning code this way is very weird, but it guess it's useful

Reply:
is it more idiomatic to split it up into functions?

Reply:
Yeah it's a  bit of a strange one, had some problems with borrows. Will try splitting it up tomorrow.